internal/bgp/community/community.go

```diff
@@ -0,0 +1,141 @@
+// SPDX-License-Identifier:Apache-2.0
+
+package community
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+// bgpCommunityType holds the type of the BGP community. Values currently supported are legacyBGPCommunity and
+// largeBGPCommunity. In the future, this can be expanded with "extendedBGPCommunity".
+type bgpCommunityType uint
+
+var (
+	ErrInvalidaCommunityValue = errors.New("invalid community value")
+	ErrInvalidCommunityFormat = errors.New("invalid community format")
+)
+
+const (
+	legacyBGPCommunity bgpCommunityType = iota
+	largeBGPCommunity
+)
+
+// largeBGPCommunityMarker is the prefix that shall be used to indicate that a given community value is of type large
+// community. If and when extended BGP communities will be supported, the largeBGPCommunityMarker allows us to
+// distinguish between extended and large communities.
+const largeBGPCommunityMarker = "large"
+
+// BGPCommunity holds the internal representation of a BGP community. It currently supports legacy and large BGP
+// communities with plans to expand this to extended communities in the future. A BGPCommunity object is intended to be
+// read only and as such all its methods directly operate on the type itself as opposed to using pointer receivers.
+type BGPCommunity struct {
+	communityType       bgpCommunityType
```

Comment:
I think a better approach would've been to have a "Community" interface and two implementations, mostly because we are already doing so, under a bunch of ifs.
That would (at least) provide a better internal representation of legacy communities, as afaiu, we are squashing the two parts under "globalAdministrator" while it's not really the same thing we'd had in large communities.

Reply:
done